client/src/accounts/authentication/authentication.hook.test.tsx

```diff
@@ -0,0 +1,37 @@
+import { ReactNode } from 'react';
+import { describe, expect, it } from 'vitest';
+import { act, renderHook, waitFor, screen } from '~/test/utils';
+import { AuthenticationProvider } from './authentication.context';
+import { useAuthentication } from './authentication.hook';
+
+interface Result {
+  children: ReactNode;
+  userId: string;
+  setUserId: Function;
+  user: string;
+  setUser: Function;
+  accessToken: string;
+  setAccessToken: Function;
+}
+
+describe('useAuthenticationHook', () => {
+  it('should include the children', async () => {
+    const expectedChildren = <h1>I am child</h1>;
+    const wrapper = () => <AuthenticationProvider>{expectedChildren}</AuthenticationProvider>;
+
+    renderHook<Result, unknown>(() => useAuthentication(), { wrapper });
+
+    act(() => {
+      expect(screen.getByRole('heading', { name: /i am child/i })).toBeInTheDocument();
+    });
```

Comment:
Expects, should never be encapsulated in `act` blocks, again, this isn't following the pattern I laid out, so I'm not going to review any further until this has been looked at.

In this one you have a wrapper a with the AuthenticationProvider, this has been added to the custom render function, so should not be necessary any longer. It is a one liner that I think makes the code harder to read than easier, sure if it needed a complex of long component hierarchy I can see doing that, but in this case, it just means you gotta jump from one bit of code to another, for one line.

Try and keep your tests consistently laid out, this is a major failing of your tests and it makes the reviewers life that bit harder, for no benefits.

Your interfact used `Function`, I'd expect the IDE to warn you against doing this, this isn't what to do, rather you should descript the function's API e.g. `setUserId: (id: string) => void`.